spec/features/admin_adds_a_group_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
+
+feature "Admin adds a group" do
+  scenario "successfully" do
+    visit new_group_path
+
+    fill_in "name", with: "St. Peter", :match => :prefer_exact
+    fill_in "responsable name", with: "Mr. Black", :match => :prefer_exact
+    fill_in "responsable contact", with: "0045-123-4445"
+    select "1", from: "group_from_date_3i"
+    select "May", from: "group_from_date_2i"
+    select "2019", from: "group_from_date_1i"
+    select "1", from: "group_to_date_3i"
+    select "May", from: "group_to_date_2i"
+    select "2019", from: "group_to_date_1i"
```

Comment:
Do you need to add a validation to the model that `:to_date` and :from_date` should be different and `:to_date` should be after `:from_date` (i.e. in the future)?

Reply:
Done with fixup 4eb0dc3.

I have a question, I tried to add some spec in validations to check when the date is wrong the Group is not valid. It didn't work, I got this error: 

`ActiveRecord::RecordInvalid:
       Validation failed: to from date can not be in the past!`

And this is my test:

```
describe "when the from date is in the past" do
    it "is not valid" do
      group = create(:group, from_date: "02/05/2016")

      expect(group).to_not be_valid
    end
  end
end
```
I suppose that I got the error when trying to create a new group so it doesn't go any further. This means that I don't need it or am I doing something wrong?

Thanks!

Reply:
You can't create it when it is not valid. Also, no reason to create and save it in the database (expensive operation).
Instead do something like:
```
context "when from_date is in the past" do
  it "is not valid" do
    group = build(:group, from_date: 1.month.ago)

    expect(group).not_to be_valid
  end
end
```

Reply:
Thanks! Done with fixup `e7def89`